pyexchange/tethfinex.py

```diff
@@ -217,43 +218,50 @@ def place_order(self,
                     pay_amount: Wad,
                     buy_token: Address,
                     buy_amount: Wad,
-                    fee_address: Address,
+                    ethfinex_address: Address,
+                    exchange_address: Address,
                     pair: str) -> Order:
 
         assert(isinstance(is_sell, bool))
         assert(isinstance(pay_token, Address))
         assert(isinstance(pay_amount, Wad))
         assert(isinstance(buy_token, Address))
         assert(isinstance(buy_amount, Wad))
-        assert(isinstance(fee_address, Address))
+        assert(isinstance(ethfinex_address, Address))
+        assert(isinstance(exchange_address, Address))
         assert(isinstance(pair, str))
+        # fee rate
+        # use 0.0025 for 0.25% fee
+        # use 0.01   for 1% fee
+        # use 0.05   for 5% fee
+        fee_rate = 0.0025
 
         expiration = int((datetime.datetime.today() + datetime.timedelta(hours=6)).strftime("%s"))
-        order = ZrxOrder(exchange=self.tethfinex,
+        order = ZrxV2Order(exchange=self.tethfinex,
+                         sender=ethfinex_address,
                          maker=Address(self.tethfinex.web3.eth.defaultAccount),
-                         taker=fee_address,
+                         taker=Address('0x0000000000000000000000000000000000000000'),
                          maker_fee=Wad(0),
                          taker_fee=Wad(0),
-                         pay_token=pay_token,
-                         pay_amount=(pay_amount / buy_amount) * buy_amount if is_sell else pay_amount,
-                         buy_token=buy_token,
-                         buy_amount=buy_amount if is_sell else (buy_amount / pay_amount) * pay_amount,
-                         salt=self.tethfinex.random_salt(),
-                         fee_recipient=fee_address,
+                         pay_asset=Asset.deserialize('0xf47261b0E' + str(pay_token)[2:]),
+                         pay_amount=pay_amount,
+                         buy_asset=Asset.deserialize('0xf47261b0E' + str(buy_token)[2:]),
+                         buy_amount=Wad.from_number(float(buy_amount) - (float(buy_amount)/100 * fee_rate * 100)),
+                         salt=self.tethfinex.generate_salt(),
+                         fee_recipient=ethfinex_address,
                          expiration=expiration,
-                         exchange_contract_address=self.tethfinex.address,
-                         ec_signature_r=None,
-                         ec_signature_s=None,
-                         ec_signature_v=None)
+                         exchange_contract_address=exchange_address,
```

Comment:
fixed

Reply:
I'm thinking to add a check when it retrieves price precision on https://api.ethfinex.com/v1/symbols_details, it catches an error when pairs don't exist, because i'm looking in it and not only DAIETH is different.